configure.py

```diff
@@ -1117,10 +1119,11 @@ def setup_first_pkg_of_list(pkglist):
 build_modes =  {m: modes[m] for m in selected_modes}
 build_artifacts = all_artifacts if not args.artifacts else args.artifacts
 
-status = subprocess.call("./SCYLLA-VERSION-GEN")
-if status != 0:
-    print('Version file generation failed')
-    sys.exit(1)
+if not args.update_version:
+    status = subprocess.call("./SCYLLA-VERSION-GEN")
```

Comment:
Maybe it is a starter's question.
But, How does the process knows when to create the files (first mode it compiles) and when not to create (not first mode) - files should exist?

Reply:
On first iteration jenkins job should call ./reloc/buid_reloc.sh, then second iteration the job should call ./reloc/build_reloc.sh --disable-update-version.
With the option, build script uses already generated SCYLLA*FILE, we can avoid date difference isue.
Note that running build_reloc.sh --disable-update-version without SCYLLA*FILE, build script will causes file not found error.

Reply:
Maybe it is better to:
If file exist, use it.
If not - create it.
So you don't need a parameter at all.
On jenkins jobs, we clean the workspace each time.
On developers machines - this is not an issue at all.

Reply:
@hagitsegev Partially agree, but I suppose we have to update version files when git hash changed (ex: user may run git checkout <branch>)
Otherwise it will be inconsistent, we cannot guarantee SCYLLA-RELEASE-FILE contains correct git hash anymore.

So I think it should be something like this:
```
if exist:
    if hash matches:
        use it
    else:
        update it
else:
    create it
```

Reply:
@hagitsegev Pushed new version based on the idea described above https://github.com/scylladb/scylla/pull/5623/files